ibis/backends/pyspark/__init__.py

```diff
@@ -514,7 +521,12 @@ def get_schema(
         table_loc = self._to_sqlglot_table((catalog, database))
         catalog, db = self._to_catalog_db_tuple(table_loc)
         with self._active_catalog_database(catalog, db):
-            df = self._session.table(table_name)
+            try:
+                df = self._session.table(table_name)
+            except AnalysisException as e:
+                if e.getErrorClass() == "TABLE_OR_VIEW_NOT_FOUND":
+                    raise com.TableNotFound(table_name) from e
```

Comment:
```suggestion
                if e.getErrorClass() == "TABLE_OR_VIEW_NOT_FOUND":
                    raise com.TableNotFound(table_name) from e
                raise
```

You still need to raise for any other `AnalysisException` that isn't handled by the error class check.

Reply:
Also this check doesn't seem to be working for PySpark 3.3.3, I'm looking into it.

Reply:
Fixed in the latest commit.